src/function/scalar/system/aggregate_export.cpp

```diff
@@ -0,0 +1,286 @@
+		bound_function.return_type = arg_return_type;
+	}
+
+	return make_unique<ExportAggregateBindData>(bound_aggr, bound_aggr.state_size());
+}
+
+static void ExportAggregateFinalize(Vector &state, FunctionData *bind_data_p, Vector &result, idx_t count,
+                                    idx_t offset) {
+	D_ASSERT(offset == 0);
+	auto bind_data = (ExportAggregateFunctionBindData *)bind_data_p;
+	auto state_size = bind_data->aggregate->function.state_size();
+	auto blob_ptr = FlatVector::GetData<string_t>(result);
+	auto addresses_ptr = FlatVector::GetData<data_ptr_t>(state);
+	for (idx_t row_idx = 0; row_idx < count; row_idx++) {
+		auto data_ptr = addresses_ptr[row_idx];
+		blob_ptr[row_idx] = StringVector::AddStringOrBlob(result, (const char *)data_ptr, state_size);
+	}
+}
+
+ExportAggregateFunctionBindData::ExportAggregateFunctionBindData(unique_ptr<Expression> aggregate_p) {
+	D_ASSERT(aggregate_p->type == ExpressionType::BOUND_AGGREGATE);
+	aggregate = unique_ptr<BoundAggregateExpression>((BoundAggregateExpression *)aggregate_p.release());
+}
+
+unique_ptr<FunctionData> ExportAggregateFunctionBindData::Copy() {
+	return make_unique<ExportAggregateFunctionBindData>(aggregate->Copy());
+}
+
+unique_ptr<BoundAggregateExpression>
+ExportAggregateFunction::Bind(unique_ptr<BoundAggregateExpression> child_aggregate) {
+	auto &bound_function = child_aggregate->function;
+	if (!bound_function.combine) {
+		throw BinderException("Cannot export state for non-combinable function %s", bound_function.name);
+	}
+	if (bound_function.bind) {
+		throw BinderException("Cannot export state on functions with custom binders");
+	}
+	if (bound_function.destructor) {
+		throw BinderException("Cannot export state on functions with destructors");
+	}
+	// this should be required
+	D_ASSERT(bound_function.state_size);
+	D_ASSERT(bound_function.finalize);
+
+	D_ASSERT(child_aggregate->function.return_type.id() != LogicalTypeId::INVALID);
+#ifdef DEBUG
+	for (auto &arg_type : child_aggregate->function.arguments) {
+		D_ASSERT(arg_type.id() != LogicalTypeId::INVALID);
+	}
+#endif
+	auto export_bind_data = make_unique<ExportAggregateFunctionBindData>(child_aggregate->Copy());
+	aggregate_state_t state_type(child_aggregate->function.name, child_aggregate->function.return_type,
+	                             child_aggregate->function.arguments);
+	auto return_type = LogicalType::AGGREGATE_STATE(move(state_type));
+
+	auto export_function =
+	    AggregateFunction("aggregate_state_export_" + bound_function.name, bound_function.arguments, return_type,
+	                      bound_function.state_size, bound_function.initialize, bound_function.update,
+	                      bound_function.combine, ExportAggregateFinalize, bound_function.simple_update,
+	                      /* can't bind this again */ nullptr, /* no dynamic state yet */ nullptr,
+	                      /* can't propagate statistics */ nullptr, bound_function.window);
```

Comment:
Can we perform custom windowing on exported aggregates?

Reply:
Probably not :)